### DISCUSSION TITLE
Do more with / better support of the Sonoff D1 Dimmer and its features

### DISCUSSION THREAD

**Discussion by jenshenk** [arendst/Tasmota#12424] [Ideas]
Hi,

After quite some tinkering around with the Sonoff D1 dimmer I'm happy that it's finally working okayish, but I was starting to think further:

- The D1 with better support could actually be much more than just a dimmer: It has an RF receiver and it has a buzzer. However, both features are currently not really controllable through tasmota afaik, and especially the rf receiver is actually creating so many  problems that people take drastic measures to make it non-functional (on forums you read about tips to crush it with a pliers among other things)
- Additionally: Since tasmota does not really control / know what the daughter board is doing simple things like getting back to the same dimmer and switch state after cutting/reconnecting power are not fully working (tasmota is doing its thing but due to unknowns about the daughter board the lights aren't really turned on). Atm the only workaround is using a rule that adjusts the light additionally after boot - this is working, however this also means that after EVERY reboot the lights will turn on, which is far from optimal. Therefore getting better support for the whole device would also help making the most basic features really work.  
-   Looking at how the sonoff RF bridge and the sonoff zigbee bridge are supported (with one part tasmota and one part special firmware so tasmota can talk to the rf/zigbee chip) I was wondering whether something similar would be possible for the D1?

- Getting more control of the rf receiver and the buzzer would be a really great feature... the d1 could become an rf bridge or "rf extender" (especially in bigger houses where you might have a d1 at a location far away from the rf bridge) and the same goes for the buzzer, which could e.g. become an alarm clock, door bell etc...

Unfortunately, my skill-set is too low to accomplish any of these ideas myself. However, I bought an extra D1 when I ordered mine and already flashed it with tasmota - I'd be happy to provide it to whomever feels like getting into this for the cost of shipping. I'd also love to contribute by beta testing. It would just be so cool to elevate this little device from barely working with workarounds to something awesome everybody could use :)

**Comment by Jason2866** [comment]
Probably there will no more time invested in the Sonoff D1 Dimmer.
Reason the Dimmer itself is not a very good one.
More info about the D1 https://github.com/arendst/Tasmota/issues/7598

**Reply by sfromis** [reply]
And Tasmota can't have access to the RF part at all. Even with replacement firmware for the other MCU, it is likely that the RF chip would not be usable as a general purpose RF receiver. This dimmer is too much trouble, seems that it doesn't even work properly with original firmware. Better ones out there. Dud product.

**Reply by jenshenk** [reply]
Thanks for your replies! Well, I cross my finger someone is still motivated after all ... seeing what people with much better skills than me in this field are capable of doing and also how many of the D1s are around there who knows, my offer to send one up still stands :)
Besides that I'm very curious which one you guys could recommend?
I've 12v LED spots behind dimmable trafos and before I got into home automation I went through a quite costly adventure of trying out multiple dumb dimmers and even pretty expensive ones were far worse than the D1... guess that's the reason why I'm still quite positive about it

**Reply by sfromis** [reply]
If you had better luck with the Sonoff D1, it is most probably not due to it being "better", just that it is the old-fashioned trailing edge dimmer type. The more modern (and generally better) leading edge dimmer type is now widespread. Trouble is just that some devices may work with one dimmer type but bad results with the other.

**Reply by jenshenk** [reply]
Thanks! That could have deffo been the reason! But since for the trafos I'm using trailing edge is recommended, the dumb dimmers were also trailing edge. What the D1 does much better than the dumb ones is switching the lights on at the lowest intensity - switching on at 1% makes the lights get a tiny bit brighter initially but they immediately go to 1%. Looks and feels nice. Dumb ones either kinda pumped the lights on at 100% (no other way to describe it ) and then went down .... with a cheaper one some of the lights did not turn on at all.
I'm curious what other smart dimmers might do even better... or do I also need to upgrade to a trafo designed for leading edge? I read about Shelly dimmers but don't know whether they are worth a try. Always happy to learn more about these things, nevertheless that it is also a bit funny that in 2021 dimming 12v leds still seems like cutting edge technology that needs a lot of research to work flawlessly.

**Reply by Jason2866** [reply]
I use a Shelly Dimmer 2. I am statisfied with. It is only a mess trying to use without neutral.
No issue for me since i have neutral everywhere i need.

**Reply by jenshenk** [reply]
Thanks for the tip, gonna give the shelly a try, too. 
For the D1: I guess I'll just wait a bit and then go the U1/U2 desoldering route to get rid of issues. I'm using a non sonoff rf433 wall switch which is ok as long as I don't power down the D1 and it is linked to the RM433 remote but now and then there are still issues.
